crypto/provider_core.c

```diff
@@ -658,7 +661,7 @@ int ossl_provider_add_to_store(OSSL_PROVIDER *prov, OSSL_PROVIDER **actualprov,
  err:
     CRYPTO_THREAD_unlock(store->lock);
     if (actualprov != NULL)
```

Comment:
The NULL check here isn't useful.

Reply:
Why do you think so? Can you please elaborate? There are NULL checks for actualprov elsewhere in the function.

Reply:
`ossl_provider_free()` does the null check and we typically just call free even if the pointer could be null.

Reply:
Please note the dereference.

Reply:
Doh!